server/src/main/java/com/aws/greengrass/cli/CLIEventStreamAgent.java

```diff
@@ -520,6 +529,99 @@ public void handleStreamEvent(EventStreamJsonMessage streamRequestEvent) {
         }
     }
 
+    class CancelLocalDeploymentHandler extends GeneratedAbstractCancelLocalDeploymentOperationHandler {
+        private final Topics cliServiceConfig;
+        private final String componentName;
+
+        public CancelLocalDeploymentHandler(OperationContinuationHandlerContext context, Topics cliServiceConfig) {
+            super(context);
+            this.cliServiceConfig = cliServiceConfig;
+            this.componentName = context.getAuthenticationData().getIdentityLabel();
+        }
+
+        @Override
+        protected void onStreamClosed() {
+
+        }
+
+        @Override
+        public CancelLocalDeploymentResponse handleRequest(CancelLocalDeploymentRequest request) {
+            return translateExceptions(() -> {
+                validateCancelLocalDeploymentRequest(request);
+                authorizeRequest(Permission.builder()
+                        .principal(componentName)
+                        .resource(AuthorizationHandler.ANY_REGEX)
+                        .operation(CANCEL_LOCAL_DEPLOYMENT)
+                        .build());
+
+                String deploymentId = request.getDeploymentId();
+                Topics localDeployments = cliServiceConfig.findTopics(PERSISTENT_LOCAL_DEPLOYMENTS);
+
+                // if deployment id cannot be found in persisted config, then return directly
+                if (localDeployments == null || localDeployments.findTopics(deploymentId) == null) {
+                    ResourceNotFoundError rnf = new ResourceNotFoundError();
+                    rnf.setMessage("Cannot find the deployment id provided " + deploymentId);
+                    rnf.setResourceType(LOCAL_DEPLOYMENT_RESOURCE);
+                    rnf.setResourceName(request.getDeploymentId());
+                    throw rnf;
+                }
+
+                // if deployment is already finished, return directly
+                Topics deploymentTopics = localDeployments.findTopics(request.getDeploymentId());
+                DeploymentStatus status =
+                        deploymentStatusFromString(Coerce.toString(deploymentTopics.find(DEPLOYMENT_STATUS_KEY_NAME)));
+                if (!DeploymentStatus.IN_PROGRESS.equals(status) && !DeploymentStatus.QUEUED.equals(status)) {
+                    CancelLocalDeploymentResponse response = new CancelLocalDeploymentResponse();
+                    response.setMessage("Cancellation request is not processed because deployment is already finished");
+                    return response;
+                }
+
+                // reuse the previous deployment id and set isCancelled=true, because deployment queue replaces
+                // queued deployments if it has the same ids.
+                Deployment deployment = new Deployment(Deployment.DeploymentType.LOCAL, deploymentId, true);
+                if (deploymentQueue == null) {
+                    logger.atError().log("Deployments queue not initialized");
```

Comment:
So we just error and do nothing? (is that queue initialization in scope of nucleus that CLI can't control?)

Reply:
Yes the queue is only managed by nucleus deployment service. CLI just offer a new deployment to the queue if possible

Reply:
updated the error message